.github/workflows/build.yml

```diff
@@ -0,0 +1,117 @@
+          bin/update-glean-documentation.sh
+          if [[   $(git diff docs/glean | head -c1 | wc -c) -ne 0 ]]; then echo "Unexpected glean changes! Please regenerate glean documentation via `./bin/update-glean-documentation.sh`." && false; fi
+
+  build-and-test-linux-base:
+    runs-on: ${{ matrix.os }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-18.04
+        python-version: [3.6, 3.7, 3.8, 3.9]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements/build.txt
+          pip install -e .
+      - name: Test
+        run: |
+          coverage run -m pytest tests
+          pip install coveralls; coveralls
+
+  build-and-test-linux-gui:
+    runs-on: ubuntu-18.04
+    env:
+      DISPLAY: ":99.0"
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: setup
+        run: |
+          sudo apt install libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0
+          /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1920x1200x24 -ac +extension GLX
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements/build.txt
+          pip install -e .
+      - name: Build
+        run: |
+          python gui/build.py bundle
+          ls -alh gui/mozregression-gui.tar.gz
+      - name: Test
+        run: |
+          coverage run -m pytest -v gui/tests
+          pip install coveralls; coveralls
+
+  build-and-test-mac-gui:
+    runs-on: macos-latest
+    env:
+      # We need the official Python, because the GA ones only support newer macOS versions
+      # The deployment target is picked up by the Python build tools automatically
+      PYTHON_VERSION: 3.9.1
+      MACOSX_DEPLOYMENT_TARGET: 10.9
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get Python
+        run: curl https://www.python.org/ftp/python/${PYTHON_VERSION}/python-${PYTHON_VERSION}-macosx10.9.pkg -o "python.pkg"
+      - name: Install Python
+        # Need to copy the pkg because the installer command removes it, breaking the cache
```

Comment:
I don't understand this comment.  Where is the copy occurring? I only see the warned-about `installer` call.

Reply:
Yeah, I stole that from here: https://github.com/actions/virtual-environments/issues/1256#issuecomment-770270252

I also don't understand the comment, all I know is that it works. 😄 I think I'll replace it with a link to ^^